File: backend/app/schemas/parent.py
import uuid

from pydantic import BaseModel, EmailStr, Field


class LinkChildRequest(BaseModel):
    student_email: EmailStr


class PermissionToggleRequest(BaseModel):
    status: str = Field(pattern="^(granted|denied)$")


class ChildSummaryResponse(BaseModel):
    student_id: uuid.UUID
    first_name: str
    last_name: str
    profile_picture_url: str | None
    link_status: str

    model_config = {"from_attributes": True}


class ParentDashboardResponse(BaseModel):
    children: list[ChildSummaryResponse] = []
    pending_permissions: int = 0

    model_config = {"from_attributes": True}

File: mobile/src/types/parent.ts
import { PermissionStatus } from './common';

export interface LinkChildRequest {
  student_email: string;
}

export interface PermissionToggleRequest {
  status: 'granted' | 'denied';
}

export interface ChildSummaryResponse {
  student_id: string;
  first_name: string;
  last_name: string;
  profile_picture_url: string | null;
  link_status: string;
}

export interface ParentDashboardResponse {
  children: ChildSummaryResponse[];
  pending_permissions: number;
}

export interface TutorPermission {
  id: string;
  tutor_id: string;
  tutor_name: string;
  status: PermissionStatus;
}

export interface ChildDetailResponse {
  student_id: string;
  first_name: string;
  last_name: string;
  profile_picture_url: string | null;
  link_status: string;
  tutor_permissions: TutorPermission[];
}
